src/message_dao.py

```diff
@@ -150,3 +169,15 @@ def store_last_hash(self, last_hash: str) -> None:
 
     def get_last_hash(self) -> int:
         return self.last_hash
+
+    def find_matching(self, normalized_subject: str = "", from_: str = "") -> List[Message]:
+        criteria = {
+            "normalized_subject": normalized_subject,
+            "from_": from_,
+        }
+    
+        def _Match(msg: Message):
+            return all(getattr(msg, attr) == value and msg.change_id is not None for attr, value in criteria.items() if value != "")
```

Comment:
we're evaluating `msg.change_id is not None` for every attribute we match on.

Perhaps instead
```python
def _Match(msg: Message):
  if msg.change_id is None:
     return False
   return all(getattr...)
```

Reply:
Makes sense; no need to check it per attribute when the return is dependent on just the message.